RevenueCatUI/Templates/V2/Components/Timeline/TimelineComponentView.swift

```diff
@@ -0,0 +1,331 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  TimelineComponentView.swift
+//
+//  Created by Mark Villacampa on 15/1/25.
+
+import Foundation
+import RevenueCat
+import SwiftUI
+
+#if PAYWALL_COMPONENTS
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct TimelineComponentView: View {
+    private let viewModel: TimelineComponentViewModel
+
+    @Environment(\.colorScheme)
+    private var colorScheme
+
+    internal init(viewModel: TimelineComponentViewModel) {
+        self.viewModel = viewModel
+    }
+
+    var body: some View {
+        VStack(alignment: .leading, spacing: viewModel.component.itemSpacing ?? 0) {
+            ForEach(viewModel.items, id: \.component) { item in
+                timelineRow(item: item)
+            }
+        }
+        // Add `itemSpacing` padding to the bottom of the timeline so the last connector can extend
+        // a little beyond the description text of the last item.
+        .padding(.bottom, viewModel.component.itemSpacing ?? 0)
+        .backgroundPreferenceValue(ItemBoundsKey.self) { bounds in
+            GeometryReader { proxy in
+                ForEach(Array(viewModel.items.enumerated()), id: \.offset) { index, item in
+                    let next = viewModel.items.indices.contains(index + 1) ? viewModel.items[index + 1] : nil
+                    // swiftlint:disable identifier_name
+                    if let from = bounds[item.id], let next, let to = bounds[next.id] {
+                        // Connect two items from center to center respecting margins
+                        if let connector = item.component.connector {
+                            let color = connector.color.asDisplayable(uiConfigProvider: viewModel.uiConfigProvider)
+                            Rectangle()
+                                .fillColorScheme(color, colorScheme: colorScheme)
+                                .frame(
+                                    width: connector.width,
+                                    height: proxy[to][.center].y - proxy[from][.center].y -
+                                    (item.component.connector?.margin.bottom ?? 0) -
+                                    (item.component.connector?.margin.top ?? 0)
```

Comment:
Hmm right... I think I will need to do some of this calculations in Android as well in cases where one item is bigger than the others 👍